packages/celer-web-app/src/app/AppDocumentProvider.tsx

```diff
@@ -48,6 +47,8 @@ export const AppDocumentProvider: React.FC<AppDocumentProviderProps> = ({ servic
 	useEffect(() => {
 		const service = serviceCreator(params);
 		service.start((doc, error, status) => {
+			// After starting the service, add it to recent pages
+			service.addToRecentPages();
```

Comment:
Should put this outside of the callback (i.e. before of after .start()). There's no contract that the callback is only called once per load. For example, the dev server calls the callback every time when the document changes. In the future we might do something like that for other routes as well